db.py

```diff
@@ -172,6 +173,51 @@ class CourseOrganization(db.Model):
     )
 
 
+class Assignment(db.Model):
```

Comment:
In all cases we don't need more than a unique table for assignments: comments and, potentially, draft fields included. A system-wide draft flag can be stored in the Configuration model.

Reply:
I don't understand why we should use the configuration table. It would mean that there would have to be a link between configuration and the line of assignments. 

That's why I'd prefer to keep the configuration table to store all the information related to the years and the assignment table would serve as a link between the common assignment lines but also to store the status of an assignment.

Reply:
Firstly, I understand the link between the configuration table and the assignment table. This would mean that for each year, there would be two configurations to store draft/published assignments.

Example for the year 2024: 

 I would therefore have two configurations, one where the assignments are drafts and one for publications. So when I publish new assignments for a configuration, the previous assignments will be replaced.

Reply:
First, I think we should rename the configuration table "Year" (or something else Anthony proposed) so it is more clear. 

I'm not sure what you are trying to explain with your example. I feel like you just need a configuration per year and the info of it being a draft inside the configuration table. You would then have an assignment table (currently AssignmentLine) where we have an entry per link between user and course-year.